### PR TITLE
Reporting the number of metrics sent and dropped for Kairos handler.

### PR DESCRIPTION
**Pull request by venkey-ariv**

### CODE REVIEW

File: src/fullerite/handler/handler.go [hunk 1]
```diff
@@ -161,3 +163,23 @@ func (handler *BaseHandler) makeEmissionTimeMetric() metric.Metric {
 func (handler *BaseHandler) resetEmissionTimes() {
 	handler.emissionTimes = make([]float64, 0)
 }
+
+func (handler *BaseHandler) makeMetricsSentMetric() metric.Metric {
+	m := metric.New("MetricsSent")
```
Comment (baris) [inline src/fullerite/handler/handler.go]: Metrics name should include Prefix()

File: src/fullerite/handler/handler.go [hunk 2]
```diff
@@ -161,3 +163,23 @@ func (handler *BaseHandler) makeEmissionTimeMetric() metric.Metric {
...
+	m.Value = float64(handler.metricsSent)
+	return m
+}
+
+func (handler *BaseHandler) resetMetricsSent() {
+	handler.metricsSent = 0
+}
+
+func (handler *BaseHandler) makeMetricsDroppedMetric() metric.Metric {
+	m := metric.New("MetricsDropped")
```
Comment (baris) [inline src/fullerite/handler/handler.go]: Prefix()

File: src/fullerite/handler/kairos.go [hunk 3]
```diff
@@ -87,22 +89,43 @@ func (k *Kairos) Run() {
 
 		if emitHandlerIntervalPassed {
 			lastHandlerMetricsEmission = time.Now()
+			// Report HandlerEmitTiming
```
Comment (baris) [inline src/fullerite/handler/kairos.go]: Looks like makeMetrics* methods are a good idea. Would you like to pull this out to base handler too?

File: src/fullerite/handler/kairos.go [hunk 4]
```diff
@@ -87,22 +89,43 @@ func (k *Kairos) Run() {
...
 			m := k.makeEmissionTimeMetric()
 			k.resetEmissionTimes()
 			m.AddDimension("handler", "Kairos")
 			datapoints = append(datapoints, k.convertToKairos(m))
+
+			// Report setrics sent
+			metricsSent := k.makeMetricsDroppedMetric()
+			k.resetMetricsSent()
+			metricsSent.AddDimension("application", "fullerite")
```
Comment (baris) [inline src/fullerite/handler/kairos.go]: Let's just add default dimensions.

File: src/fullerite/handler/kairos.go [hunk 5]
```diff
@@ -87,22 +89,43 @@ func (k *Kairos) Run() {
...
+			metricsSent.AddDimension("handler", "kairos")
```
Comment (baris) [inline src/fullerite/handler/kairos.go]: makeMetric* functions should do this. Something like: AddDimension("handler", h.Name()).

File: src/fullerite/handler/kairos.go [hunk 6]
```diff
@@ -87,22 +89,43 @@ func (k *Kairos) Run() {
...
-			k.emitMetrics(datapoints)
+			result := k.emitMetrics(datapoints)
 			lastEmission = time.Now()
 
 			emissionTimeInSeconds := lastEmission.Sub(beforeEmission).Seconds()
 			k.log.Info("POST to Kairos took ", emissionTimeInSeconds, " seconds")
 			k.emissionTimes = append(k.emissionTimes, emissionTimeInSeconds)
 
+			if result {
+					k.metricsSent += len(datapoints)
```
Comment (baris) [inline src/fullerite/handler/kairos.go]: indentation

File: src/fullerite/handler/handler.go (hunk 1 above)
Comment (venkey-ariv) [inline src/fullerite/handler/handler.go]: Done. I changed this to handler.Prefix()+"MetricsSent"

File: src/fullerite/handler/handler.go (hunk 2 above)
Comment (venkey-ariv) [inline src/fullerite/handler/handler.go]: Same as above

File: src/fullerite/handler/kairos.go (hunk 3 above)
Comment (venkey-ariv) [inline src/fullerite/handler/kairos.go]: Done

File: src/fullerite/handler/kairos.go (hunk 4 above)
Comment (venkey-ariv) [inline src/fullerite/handler/kairos.go]: Done

**Pull request merged**